README.md

```diff
@@ -13,207 +13,111 @@ Istanbul's state of the art command line interface, with support for:
-```
+See `npx nyc --help` for all options available.
+You can set these in any of the files listed above, or from the command line.
+This table is a quick TLDR for the rest of this readme and there are more advanced docs available.
 
-To view your coverage report:
+| Option name | Description | Type | Default |
+| ----------- | ----------- | ---- | ------- |
+| `all` | Whether or not to instrument all files (not just the ones touched by your test suite) | `Boolean` | `false` |
+| `check-coverage` | Check whether coverage is within thresholds, fail if not | `Boolean` | `false` |
+| `extension` | List of extensions that nyc should attempt to handle in addition to `.js` | `Array<String>` | `['.js']` |
+| `include` | See [selecting files for coverage] for more info | `Array<String>` | `['**']`|
+| `exclude` | See [selecting files for coverage] for more info | `Array<String>` | [list](https://github.com/istanbuljs/istanbuljs/blob/master/packages/test-exclude/index.js#L176-L184) |
+| `reporter` | [Coverage reporters to use](https://istanbul.js.org/docs/advanced/alternative-reporters/) | `Array<String>` | `['text']` |
+| `report-dir` | Where to put the coverage report files | `String` | `./coverage` |
+| `skip-full` | Don't show files with 100% statement, branch, and function coverage | `Boolean` | `false` |
+| `temp-dir` | Directory to output raw coverage information to | `String` | `./.nyc_output` |
 
-<img width="500" src="screen.png">
+Configuration can also be provided by `nyc.config.js` if programmed logic is required:
+```js
+'use strict'; 
+const {defaultExclude} = require('test-exclude');
+const isWindows = require('is-windows');
 
-You can use [any reporters that are supported by `istanbul`](https://github.com/istanbuljs/istanbuljs/tree/master/packages/istanbul-reports/lib): `clover`, `cobertura`, `html`, `json-summary`, `json`, `lcov`, `lcovonly`, `none`, `teamcity`, `text-lcov`, `text-summary`, `text`.
+let platformExclude = [
+  isWindows() ? 'lib/posix.js' : 'lib/win32.js'
+];
 
-```bash
-nyc report --reporter=lcov
+module.exports = {
+  exclude: platformExclude.concat(defaultExclude)
+};
 ```
 
-You can find examples of the output for various reporters [here](https://istanbul.js.org/docs/advanced/alternative-reporters).
+### Publish and reuse your nyc configuration
 
-You also have the choice of using a [custom reporter](https://github.com/pedrocarrico/istanbul-reporter-aws-cloudwatch-metrics).
-Install custom reporters as a development dependency and you can use the `--reporter` flag to load and view them:
+nyc allows you to inherit other configurations using the key `extends`.
+As an example, an alternative way to configure nyc for `babel-plugin-istanbul` would be to use the [@istanbuljs/nyc-config-babel preset](https://www.npmjs.com/package/@istanbuljs/nyc-config-babel):
 
-```bash
-nyc report --reporter=<custom-reporter-name>
+```json
+{
+  "nyc": {
+    "extends": "@istanbuljs/nyc-config-babel"
+    //add your custom overrides here
```

Comment:
If we make this snippet `.js` rather than `.json`, it will look less wonky.

Reply:
Will do thanks